ibis/pandas/core.py

```diff
@@ -237,24 +286,27 @@ def execute_until_in_scope(
     if op in new_scope:
         return new_scope
 
-    # figure out what arguments we're able to compute on based on the
-    # expressions inputs. things like expressions, None, and scalar types are
-    # computable whereas ``list``s are not
-    computable_args = [arg for arg in op.inputs if is_computable_input(arg)]
-
     # recursively compute each node's arguments until we've changed type
+
+    if len(arg_timecontexts) != len(computable_args):
+        raise com.IbisError(
+            'arg_timecontexts differ with computable_arg in length '
+            'for type:\n{}.'.format(type(op).__name__)
```

Comment:
you can use f-strings

Reply:
I removed this check to ```another canonicalize_context``` function and use f-strings in error handling.